### PR TITLE
[mesheryctl] docker client init update

### LINKED ISSUE #7594: [mesheryctl] Docker: False negative `system start` message
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
We have a high priority bug surfacing in `mesheryctl system start` having to do with recent changes to interaction with Docker. The following error occurs despite Docker being locally available.

`system start` executes and does start the Meshery deployment, but reports a misleading error to users, whereas it should instead open their default web browser to the UI.

#### Screenshots/Logs

```
➜  ~ mesheryctl system start
Updating Meshery now...
[+] Running 12/12
 ⠿ meshery-cilium Pulled                                                                                                                                              1.4s
 ⠿ meshery-nsm Pulled                                                                                                                                                 1.5s
 ⠿ meshery-app-mesh Pulled                                                                                                                                            1.6s
 ⠿ meshery-traefik-mesh Pulled                                                                                                                                        1.6s
 ⠿ meshery-linkerd Pulled                                                                                                                                             1.6s
 ⠿ meshery-nginx-sm Pulled                                                                                                                                            1.6s
 ⠿ meshery-istio Pulled                                                                                                                                               1.6s
 ⠿ meshery-consul Pulled                                                                                                                                              1.4s
 ⠿ watchtower Pulled                                                                                                                                                  1.6s
 ⠿ meshery Pulled                                                                                                                                                     1.6s
 ⠿ meshery-osm Pulled                                                                                                                                                 1.6s
 ⠿ meshery-kuma Pulled                                                                                                                                                1.6s
Starting Meshery...
[+] Running 12/0
 ⠿ Container meshery-meshery-istio-1         Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-linkerd-1       Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-nsm-1           Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-consul-1        Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-osm-1           Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-kuma-1          Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-nginx-sm-1      Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-app-mesh-1      Running                                                                                                                  0.0s
 ⠿ Container meshery-watchtower-1            Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-cilium-1        Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-1               Running                                                                                                                  0.0s
 ⠿ Container meshery-meshery-traefik-mesh-1  Running                                                                                                                  0.0s
Error: failed to start Meshery
See https://docs.meshery.io/reference/mesheryctl/system for usage details
: failed to fetch the list of containers
See https://docs.meshery.io/reference/mesheryctl/system for usage details
: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

```
➜  ~ mesheryctl system context view --all
local:
  endpoint: http://localhost:9081
  token: default
  token-location: auth.json
  platform: docker
  components:
  - meshery-app-mesh
  - meshery-istio
  - meshery-linkerd
  - meshery-consul
  - meshery-nsm
  - meshery-kuma
  - meshery-osm
  - meshery-traefik-mesh
  - meshery-nginx-sm
  - meshery-cilium
  channel: stable
  version: latest
  provider: Meshery
```

#### Environment

- Host OS: Mac
- Meshery Server Version: stable-v0.6.84	304a7b6f
- Meshery Client Version: stable-v0.6.84	304a7b6f
- Platform: Docker Desktop for Mac - version 4.17.0 (99724)

[Slack Message](https://layer5io.slack.com/archives/C0130BQ4L3T/p1682207581917329)

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Azanul**
**Notes for Reviewers**

This PR fixes #7594 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/system/start.go [hunk 1]
```diff
@@ -311,8 +314,14 @@ func start() error {
 
 		checkFlag := 0 //flag to check
 
+		// Get the Docker configuration
+		dockerCfg, err := cliconfig.Load(dockerconfig.Dir())
+		if err != nil {
+			return errors.Wrap(err, utils.SystemError("failed to load config file"))
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/start.go:320]: @Azanul  will you have this turned into a MeshKit-style error, complete with probable cause, error code, and suggested remediation?

Comment (Azanul) [reply mesheryctl/internal/cli/root/system/start.go]: Check it out

**Comment by Chadha93** [comment]
@Azanul  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

File: mesheryctl/internal/cli/root/system/error.go [hunk 2]
```diff
@@ -146,3 +147,13 @@ func ErrFailedGetEphemeralPort(err error) error {
...
+
+func ErrCreatingDockerClient(err error) error {
+	return errors.New(
+		ErrCreatingDockerClientCode,
+		errors.Critical,
+		[]string{"Failed to create Docker client"},
+		[]string{"Error occurred while creating Docker client from config file", err.Error()},
+		[]string{"Missing or invalid docker config"},
+		[]string{"Please check the Docker config file for any errors or missing information. Make sure it is correctly formatted and contains all the required fields."})
+}
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/error.go:159]: @Azanul there's nothing to mention about docker contexts, huh?

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/error.go:159]: Thanks. What if we attempt to use an invalid context? What if they have no valid context? Users don't need to confirm or verify their contexts, huh?

Comment (leecalcote) [reply mesheryctl/internal/cli/root/system/error.go:159]: mesheryctl doesn't inform the user of the attempted use of a context and the state of that failure?

Comment (Azanul) [reply mesheryctl/internal/cli/root/system/error.go:159]: In case of invalid context his `docker ps` won't work either

**Review by leecalcote — APPROVED** [review]

**Pull request merged**